ballerina/tests/csv_io.bal

```diff
@@ -99,6 +113,87 @@ isolated function testReadCsv() returns Error? {
     check csvChannel.close();
 }
 
+@test:Config {}
+isolated function testReadCsvRecord() returns Error? {
+    string filePath = RESOURCES_BASE_PATH + "datafiles/io/records/sample5.csv";
+    string[][] input = check fileReadCsv(filePath);
+    string[][] expected = [["User1", "WSO2", "10000.50"],["User2", "WSO2", "20000.50"],["User3", "WSO2", "30000.00"]];
+    test:assertEquals(input[0][0], "User1");
+    test:assertEquals(input[0][2], " 10000.50");
+    Employee[] input2 = check fileReadCsv(filePath);
+    test:assertEquals(input2[0].id, "User1");
+    test:assertEquals(input2[0].salary, 10000.50f);
+
+}
+
+@test:Config {}
+function testReadFileCsvAsStreamUsingResourceFile() returns error? {
+    string filePath = RESOURCES_BASE_PATH + "datafiles/io/records/sample5.csv";
+    string[][] expected = [["User1", "WSO2", "10000.50"],["User2", "WSO2", "20000.50"],["User3", "WSO2", "30000.00"]];
+    stream<string[], Error?> result = check fileReadCsvAsStream(filePath);
+    int i = 0;
+    check result.forEach(function(string[] val) {
+        int j = 0;
+        foreach string s in val {
+            test:assertEquals('string:trim(s), 'string:trim(expected[i][j]));
+            j += 1;
+        }
+        i += 1;
+    });
+    test:assertEquals(i, 3);
+}
+
+@test:Config {}
+function testReadFileCsvAsStreamUsingResourceFileRecord() returns error? {
+    string filePath = RESOURCES_BASE_PATH + "datafiles/io/records/sample5.csv";
+    string[] expected = ["10000.50","20000.50","30000.00"];
+    stream< Employee4, Error?> result = check fileReadCsvAsStream(filePath);
+    int i = 0;
+    check result.forEach(function(Employee4 val) {
+        test:assertEquals(val.salary, expected[i]);
+        i=i+1;
+    });
+    test:assertEquals(i, 3);
+}
+
+
+@test:Config {}
+isolated function testWriteRecordCsv() returns Error? {
+    Employee E ={
+    id:"1",
+    name:"Foo",
+    salary:300000.0f
+    };
+    Employee B ={
+    id:"1",
+    name:"Foo",
+    salary:300000.0f
+    };
+
+
+    string filePath = TEMP_DIR + "recordsDefault_records.csv";
+    Employee[] content1 = [E,B];
+    test:assertEquals(check fileWriteCsv(filePath,content1),()) ;
```

Comment:
you need to check the csv file content and verify whether values are written properly

Reply:
Done.